### PR TITLE
initial code

### PR DESCRIPTION
**Pull request by kishkulk**
Initial set of code for libjansson

### CODE REVIEW

File: buildenv [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+zopen_append_to_env()
+{
+cat <<ZZ
+if [ ! -z "\$ZOPEN_IN_ZOPEN_BUILD" ]; then
+  export ZOPEN_EXTRA_CFLAGS="\${ZOPEN_EXTRA_CFLAGS} -I\$PWD/include"
+  export ZOPEN_EXTRA_CXXFLAGS="\${ZOPEN_EXTRA_CXXFLAGS} -I\$PWD/include"
+  export ZOPEN_EXTRA_LDFLAGS="\${ZOPEN_EXTRA_LDFLAGS} -L\$PWD/lib"
+  export ZOPEN_EXTRA_LIBS="\${ZOPEN_EXTRA_LIBS} -ljansson \$PWD/lib/libjansson.a"
```
Comment (IgorTodorovskiIBM) [inline buildenv]: Is \$PWD/lib/libjansson.a necessary?

Comment (kishkulk) [reply buildenv]: @IgorTodorovskiIBM removed \$PWD/lib/libjansson.a

**Comment by IgorTodorovskiIBM** [comment]
@kishkulk - there's a couple of conflicts here. Let me know if you're able to look into it

File: buildenv [hunk 2]
```diff
@@ -1,26 +1,29 @@
-JANSSON_VERSION="2.14"
+export JANSSON_VERSION="2.14"
 export ZOPEN_GIT_URL="https://github.com/akheron/jansson.git"
-export ZOPEN_GIT_DEPS="git make m4 autoconf perl libtool automake bash zoslib tar sed coreutils"
+export ZOPEN_GIT_DEPS="git make cmake m4 autoconf perl libtool automake bash zoslib tar sed coreutils"
 export ZOPEN_TYPE="GIT"
 export ZOPEN_RUNTIME_DEPS=""
 export ZOPEN_COMP="clang"
+export ZOPEN_BOOTSTRAP='./autogen'
+export ZOPEN_EXTRA_CFLAGS="-mzos-target=zosv2r4 -D_ALL_SOURCE -D_OPEN_SYS_FILE_EXT=1 -D_AE_BIMODAL=1"
```
Comment (IgorTodorovskiIBM) [inline buildenv]: Does it compile without these extra cflags?

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM, with one question

File: buildenv (hunk 2 above)
Comment (kishkulk) [reply buildenv]: It does compile without the extra cflags, did the changes

**Pull request merged**